main.cpp

```diff
@@ -28,510 +28,592 @@ Thing: Car Wash
-    void changeCutoff(double cutoffFrequency);
-//    2) change oscillator resonance
-    void changeResonance(double resonanceFrequency);
-//    3) receive host automation
-    void getAutomationFromHost(int parameterName);
+    double changeCutoff(double cutoffFrequency);
+    double changeResonance(double resonanceFrequency);
+    int getAutomationFromHost(int parameterNumber);
 };
 
-/*
-Thing 8) Amplitude Envelope
-5 properties:
-    1) Attack value
-    2) Decay value
-    3) Sustain value
-    4) Release value
-    5) Hold value
-3 things it can do:
-    1) adjust oscillator attack
-    2) receive host automation
-    3) change dry/wet mix value
- */
+double Filter::changeCutoff(double cutoff)
+{
+    return cutoff;   
+}
+
+double Filter::changeResonance(double resonance)
+{
+    return resonance;
+}
+
+int Filter::getAutomationFromHost(int parameterNumber)
+{
+    return parameterNumber;
+}
 
 struct AmplitudeEnvelope
 {
-//    1) Attack value
     int attack = 34;
-//    2) Decay value
     int decay = 45;
-//    3) Sustain value
     int sustain = 100;
-//    4) Release value
     int release = 20;
-//    5) Hold value
     int hold = 0;
 
-//    1) adjust oscillator attack
-    void changeAttack(int attack);
-//    2) receive host automation
-    void getAutomationFromHost(int parameterName);
-//    3) change dry/wet mix value
-    void changeDryWetMix(int dryValue, int wetValue);
-
+    void changeAttack(int newAttack);
+    int getAutomationFromHost(int parameterNumber);
+    int changeDryWetMix(int wetValue);
 };
 
-/*
-Thing 9) Arpeggiator
-5 properties:
-    1) Octave range value
-    2) Number of note repetitions before moving to next note
-    3) sequence pattern number
-    4) note playing direction
-    5) pitch duration selection
-3 things it can do:
-    1) be bypassed
-    2) play held notes
-    3) play a certain note duration
- */
+void AmplitudeEnvelope::changeAttack(int newAttack)
+{
+    attack = newAttack;
+}
+
+int AmplitudeEnvelope::getAutomationFromHost(int parameterNumber)
+{
+    return parameterNumber;
+}
+
+int AmplitudeEnvelope::changeDryWetMix(int wetValue)
+{
+    int dryValue = 1;
+    int mix = dryValue - wetValue;
+    return mix;
+}
 
 struct Arpeggiator
 {
-//    1) Octave range value
     int octaveSetting = 0;
-//    2) Number of note repetitions before moving to next note
     int noteRepetitions = 2;
-//    3) sequence pattern number
     int sequencePattern = 3;
-//    4) note playing direction
     std::string arpDirection = "Up/Down";
-//    5) pitch duration selection
     int pitchDuration = 16;
 
 
-//    1) be bypassed
-    void bypassArp(bool ArpIsOn);
-//    2) play held notes
-    void playHeldNotes(int noteMidiNumbers); // would need to be array eventually
-//    3) play a certain note duration
-    int playNoteDuration(int noteDuration);
-};
+    void bypassArp(bool arpIsOn)
+    {
+        if(!arpIsOn)
+            arpIsOn = false;
+        else
+            arpIsOn = true;
```

Comment:
You aren't changing the value of `arpIsOn` in these expressions.
your first 2 lines say: if `arpIsOn` is `false`, assign `false` to `arpIsOn`
- [ ] revise

Reply:
Done.